primefaces/src/main/resources/META-INF/resources/primefaces/core/core.ajax.js

```diff
@@ -148,6 +148,32 @@ if (!PrimeFaces.ajax) {
                 return null;
             },
 
+            /**
+             * Get source ID from settings.
+             *
+             * @param {JQuery.AjaxSettings} settings containing source ID.
+             * @return {string} The source ID from settings or `null` if settings does not contain a source.
+             */
+            getSourceId: function(settings) {
+                if (settings && settings.source) {
+                    return typeof settings.source === 'string' ? settings.source : settings.source.name;
+                }
+                return null;
+            },
+
+            /**
+             * Checks whether the component ID from the provided configuration equals the source ID from the provided
+             * settings.
+             *
+             * @param {PrimeFaces.PartialWidgetCfg<TCfg>} cfg of the component to check for being the source.
+             * @param {JQuery.AjaxSettings} settings containing source ID.
+             * @returns {boolean} `true` if the component ID from the provided configuration equals the source ID from
+             * the provided settings.
+             */
+            isXhrSource: function(cfg, settings) {
```

Comment:
i would either pass the id or the whole widget

Reply:
not sure ;D i dont see any reason to pass a wrapper object from the middle of the stack, isntead of the whole widget or only the id